src/meili/map_matcher.cc

```diff
@@ -309,30 +309,45 @@ MapMatcher::OfflineMatch(
     const std::vector<Measurement>& measurements, uint32_t k) {
   mapmatching_.Clear();
 
-  const auto begin = measurements.begin(),
-               end = measurements.end();
-
-  if (begin == end) {
+  if(measurements.empty())
     return {};
-  }
 
-  const auto max_search_radius = config_.get<float>("max_search_radius"),
+  const float max_search_radius = config_.get<float>("max_search_radius"),
           sq_max_search_radius = max_search_radius * max_search_radius;
-  const auto interpolation_distance = config_.get<float>("interpolation_distance"),
+  const float interpolation_distance = config_.get<float>("interpolation_distance"),
           sq_interpolation_distance = interpolation_distance * interpolation_distance;
-  std::unordered_map<StateId::Time, std::vector<Measurement>> interpolated_measurements;
+  std::unordered_map<StateId::Time, std::vector<Measurement>> interpolated;
 
   // Always match the first measurement
-  auto time = AppendMeasurement(*begin, sq_max_search_radius);
-  auto latest_match_measurement = begin;
-  for (auto measurement = std::next(begin); measurement != end; measurement++) {
-    const auto sq_distance = GreatCircleDistanceSquared(*latest_match_measurement, *measurement);
-    // Always match the last measurement
-    if (sq_interpolation_distance < sq_distance || std::next(measurement) == end) {
-      time = AppendMeasurement(*measurement, sq_max_search_radius);
-      latest_match_measurement = measurement;
-    } else {
-      interpolated_measurements[time].push_back(*measurement);
+  auto last = measurements.cbegin();
+  auto time = AppendMeasurement(*last, sq_max_search_radius);
+  double interpolated_epoch_time = -1;
+  for (auto m = std::next(last); m != measurements.end(); ++m) {
+    const auto sq_distance = GreatCircleDistanceSquared(*last, *m);
+    // Always match the last measurement and if its far enough away
+    if (sq_interpolation_distance < sq_distance || std::next(m) == measurements.end()) {
+      // If there were interpolated points between these two points with time information
+      if(interpolated_epoch_time != -1) {
+        // Project the last interpolated point onto the line between the two match points
+        auto p = interpolated[time].back().lnglat().Project(last->lnglat(), m->lnglat());
+        // If its significantly closer to the previous match point then it looks like the trace lingered
+        // so we use the time information of the last interpolation point as the actual time they started
+        // traveling towards the next match point which will help us determine what paths are really likely
+        if(p.Distance(last->lnglat())/last->lnglat().Distance(m->lnglat()) < .2f)
```

Comment:
this `.2f` is a magic number. i can make it a constant or we can make it configurable, what do @ptpt and @dnesbitt61 think?